### ISSUE TITLE
Ltac2: strange syntax error with Notation only after retracting Ltac2 in CoqIDE/PG

### ISSUE THREAD

**Issue by MSoegtropIMC** [coq/coq#11793]
#### Description of the problem

I get a syntax error on the definition of Ltac2 notations, but only after certain edit operations in CoqIDE. If the same code is run from scratch, it works fine.

The code and instructions for reproduction (what to edit) are included in the
 attached file. For me these steps a reproducible (5/5).

[ltac2_notations.txt](https://github.com/coq/coq/files/4312574/ltac2_notations.txt)

#### Coq Version

The Coq Proof Assistant, version 8.11.0 (February 2020)
compiled on Feb 12 2020 10:05:40 with OCaml 4.07.1
OSX Catalina

**Comment by whonore** [comment]
It seems to happen the second time Ltac2 is imported. I can reproduce the error with this smaller version:
```
Print nat. (* Needed to prevent CoqIDE from restarting at the start of the file *)
Require Import Ltac2.Ltac2.
Ltac2 Notation "eval_cbv" s(strategy) c(tactic(0)) :=
  Std.eval_cbv s c.

(*
Go to end.
Step backwards twice.
Go to end.
=> Error around tactic notations
Syntax error: [input_fun] expected after [binder] (in [tac2def_body]).
*)

```

**Comment by kyoDralliam** [comment]
FYI it also happens with proof general in emacs: it is enough to step through `From Ltac2 Require Import Ltac2.` then retract it and re-step a second time to prevent any Ltac2 Notation from working.

**Comment by jfehrle** [comment]
#12575 describes the root cause.

**Comment by Zimmi48** [comment]
Why keep this issue open if it is the same as #12575?

**Comment by MSoegtropIMC** [comment]
Usually one keeps the older one of two duplicates, but since #12575 is much more informative, I will close this.

**Comment by SkySkimmer** [comment]
This appears not to be a duplicate of #12575 as eg 
~~~coq
Require Import Ltac2.Ltac2.

Ltac2 Notation "dbg" := ().
~~~
running twice fails in 8.16 but works in master (since 89ad33f apparently), but #12575 still reproduces in master.

**Comment by Blaisorblade** [comment]
Copied the milestone from https://github.com/coq/coq/pull/16946 based on SkySkimmer's comment.
